From 1b629ded094c800293cf694e85d0c5b66cc9fcfd Mon Sep 17 00:00:00 2001
From: mokopatches <mokopatches@openmoko.org>
Date: Fri, 25 Jul 2008 22:21:25 +0100
Subject: [PATCH] fix-hwecc-2410.patch
 S3C24xx ECC mis-calculates the bit to flip:
 http://lists.infradead.org/pipermail/linux-mtd/2007-October/019586.html
 If the error couldn't be corrected, we returned "no problem" :-(
 http://lists.infradead.org/pipermail/linux-mtd/2007-October/019615.html

Signed-off-by: Werner Almesberger <werner@openmoko.org>
---
 drivers/mtd/nand/s3c2410.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/mtd/nand/s3c2410.c b/drivers/mtd/nand/s3c2410.c
index 8e1e482..4f747b1 100644
--- a/drivers/mtd/nand/s3c2410.c
+++ b/drivers/mtd/nand/s3c2410.c
@@ -406,7 +406,7 @@ static int s3c2410_nand_correct_data(struct mtd_info *mtd, u_char *dat,
 	if ((diff0 & ~(1<<fls(diff0))) == 0)
 		return 1;
 
-	return -1;
+	return -EBADMSG;
 }
 
 /* ECC functions
-- 
1.5.6.3

